src/node.cc

```diff
@@ -2849,58 +2847,103 @@ void ProcessEmitWarning(Environment* env, const char* fmt, ...) {
   f.As<v8::Function>()->Call(process, 1, &arg);
 }
 
+static bool pullFromCache(Environment* env,
+                          const FunctionCallbackInfo<Value>& args,
+                          Local<String> module,
+                          Local<Object> cache) {
+  Local<Object> exports;
+  if (cache->Has(env->context(), module).FromJust()) {
+    exports = cache->Get(module)->ToObject(env->isolate());
+    args.GetReturnValue().Set(exports);
+    return true;
+  }
+  return false;
+}
+
+static Local<Object> init_module(Environment* env,
+                                 node_module* mod,
+                                 Local<String> module) {
+  Local<Object> exports = Object::New(env->isolate());
+  // Internal bindings don't have a "module" object, only exports.
+  CHECK_EQ(mod->nm_register_func, nullptr);
+  CHECK_NE(mod->nm_context_register_func, nullptr);
+  Local<Value> unused = Undefined(env->isolate());
+  mod->nm_context_register_func(exports, unused,
+    env->context(), mod->nm_priv);
+  return exports;
+}
+
+static void no_such_module(Environment* env, const char* module_v) {
```

Comment:
`InitModule()` and `NoSuchModule()` would be more idiomatic names (although the latter should really indicate that it throws an exception.)

Reply:
done